.github/workflows/publish.yml

```diff
@@ -11,23 +11,38 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
-  repository_dispatch:
-    types: [create-release]
-    client_payload:
+  workflow_dispatch:
+    inputs:
       release_tag:
         required: true
+        description: 'Tag for the new release'
+    prerelease:
+      description: 'If true, the release will be set as a prerelease and not impact the latest tag'
+      type: boolean
+      required: true
+      default: false
+
 
 jobs:
   create_release:
-    if: github.event_name == 'repository_dispatch'
+    if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
+    outputs:
+      release_id: ${{ steps.create.outputs.release_id }}
     steps:
       - name: Create release
         env:
           RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+          PRERELEASE: ${{ github.events.input.prerelease }}
           GH_TOKEN: ${{ github.token }}
+        shell: pwsh
         run: |
-          gh release create "$RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
+          $prereleaseFlag = if ($env:PRERELEASE -eq $true) { '--prerelease' } else { '' }
+          $releaseUrl = gh release create "$env:RELEASE_TAG" $prereleaseFlag --draft --generate-notes --title "Release $env:RELEASE_TAG"
```

Comment:
This won't work as you expected. When you pass an empty string to a program on the command line, it's still recognized as an extra argument, which will usually mess up argument passing (when you really want the program to ignore the argument).

PowerShell has some hacks though:
```suggestion
          $prereleaseFlag = if ($env:PRERELEASE -eq $true) { '--prerelease' } else { '' }
          $releaseUrl = gh release create "$env:RELEASE_TAG" @prereleaseFlag --draft --generate-notes --title "Release $env:RELEASE_TAG"
```
`@variable` split the variable into separate arguments and pass them into the program. It's really meant for use with arrays and hash maps, but it can also work for strings because of weak typing. When `$variable` is empty, `@variable` becomes an empty array i.e. no argument at all.
See docs on splatting: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting

To do this properly though, best to construct dynamic arguments as an array rather than a string.
Also, GitHub Actions inputs and environment variables are all ultimately strings. `-eq $true` kinda works because PowerShell has weak typing, but still best to use `-eq 'true'`
```suggestion
          $prereleaseFlag = if ($env:PRERELEASE -eq 'true') { @('--prerelease') } else { @() }
          $releaseUrl = gh release create "$env:RELEASE_TAG" @prereleaseFlag --draft --generate-notes --title "Release $env:RELEASE_TAG"
```

Reply:
Hey this is a cool trick. I've looked at the docs and tried this in my own powershell and can't get it working. 

Using `$a = if ($PRERELEASE -eq 'true') { @('--prerelease') } else { @() }` I get an error `The error message is: CreateFile -: The system cannot find the file specified.` which makes me think for some reason the github command thinks that argument is a file path? 

The weird thing though is if I do `$a = @('--prerelease')` that works fine and creates the release. I'm confused that somehow the if statement is causing a problem for the github command. 

The line I'm using to try creating a release is ` $releaseUrl = gh release create "vtest" --draft @a`

Reply:
I tried using a hashtable to get around this but then the formatting is bad `bad flag syntax: --prerelease:True`

Reply:
Thought I could do some flashy line but maybe it's just easier if I do a standard if else block to check the value of `$env:PRERELEASE`

Reply:
yeah regular if else block please

Reply:
🤝